src/goto-instrument/document_properties.cpp

```diff
@@ -238,7 +238,7 @@ document_propertiest::get_code(const source_locationt &source_location)
       while(line_no.size()<4)
         line_no=" "+line_no;
 
-      line_no+"  ";
+      line_no += "  ";
```

Comment:
Seems unrelated?!

Reply:
I agree that this is worth separating into its own PR. Some of the new versions of the standard library add the `[[nodiscard]]` attribute to the + operator of `std::string`. This causes a compile error when the result of the concatenation is unused. It essentially promotes a logic error into a compile error. Without this commit, the `check-ubuntu-22_04-cmake-gcc-13` job will fail, so it does appear to be needed before we can switch standards. However as it is a logic error currently we could fix it before switching standards.

Reply:
That is a fix for an issue brought up by the C++17 change as the `operator+` between strings is marked `nodiscard` and so it was resulting in a compilation error.

If it is too awkward to include this here, I'm happy to move the change to a separate PR, but I would need it in before upgrading the standard.

Reply:
I think this should be PR'ed and fixed right away. (I guess no one is using that feature so it went unnoticed, but it should really be fixed now.)

Reply:
@tautschnig I just read the surrounding code and the value of `line_no` is never read back and output. So it is just as broken before and after this commit...

Reply:
Fair point. I suspect this should have been inserted as a prefix into each line of output. Let me make a stab at a PR.

Reply:
Fix in #7994.

Reply:
I will remove this commit from this PR as soon as the latter is merged.